sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/ProactiveOpenConnectionsProcessor.java

```diff
@@ -0,0 +1,253 @@
+            .parallel(concurrencyConfiguration.openConnectionOperationEmissionConcurrency)
+            .runOn(CosmosSchedulers.OPEN_CONNECTIONS_BOUNDED_ELASTIC)
+            .flatMap(openConnectionTask -> {
+                Uri addressUri = openConnectionTask.getAddressUri();
+                URI serviceEndpoint = openConnectionTask.getServiceEndpoint();
+                int minConnectionsForEndpoint = openConnectionTask.getMinConnectionsRequiredForEndpoint();
+                String collectionRid = openConnectionTask.getCollectionRid();
+
+                return Flux.zip(Mono.just(openConnectionTask), openConnectionsHandler.openConnections(
+                        collectionRid,
+                        serviceEndpoint,
+                        Arrays.asList(addressUri),
+                        this,
+                        minConnectionsForEndpoint));
+            }, false, concurrencyConfiguration.openConnectionExecutionConcurrency)
+            .flatMap(openConnectionTaskToResponse -> {
+                OpenConnectionTask openConnectionTask = openConnectionTaskToResponse.getT1();
+                OpenConnectionResponse openConnectionResponse = openConnectionTaskToResponse.getT2();
+
+                if (openConnectionResponse.isConnected() && openConnectionResponse.isOpenConnectionAttempted()) {
+                    this.submitOpenConnectionWithinLoopInternal(openConnectionTask);
+                    return Mono.just(openConnectionResponse);
+                }
+
+                // open connections handler has created the min. required connections for the endpoint
+                if (openConnectionResponse.isConnected() && !openConnectionResponse.isOpenConnectionAttempted()) {
+                    // for the specific service endpoint, it has met the mini pool size requirements, so remove from the map
+                    // TODO: response should reflect all previous opened connection
+                    this.removeEndpointFromMonitor(openConnectionTask.getAddressUri().toString(), openConnectionResponse);
+                    return Mono.just(openConnectionResponse);
+                }
+
+                return openConnectionTask
+                        .getRetryPolicy()
+                        .shouldRetry((Exception) openConnectionResponse.getException())
+                        .flatMap(shouldRetryResult -> {
+                            if (shouldRetryResult.shouldRetry) {
+                                return enqueueOpenConnectionOpsForRetry(openConnectionTask, shouldRetryResult);
+                            }
+
+                            // TODO: response should reflect all previous opened connection
+                            this.removeEndpointFromMonitor(openConnectionTask.getAddressUri().toString(), openConnectionResponse);
+                            return Mono.just(openConnectionResponse);
+                        });
+            })
+            .runOn(CosmosSchedulers.OPEN_CONNECTIONS_BOUNDED_ELASTIC)
```

Comment:
We already have this above on line 134 for this call, do we need this twice?

Reply:
Redundant operator usage. Removed it.